### ISSUE TITLE
Tableau Connector : Unify Data Models

### ISSUE THREAD

**Issue by jsampaiog** [open-metadata/OpenMetadata#15218]
**Is your feature request related to a problem? Please describe.**
When ingesting Data Models in tableau, multiple datamodels are displayed for the same data source.
This explodes the number of total data sources, even though unique, and makes discovery and lineage more complicated.
![image](https://github.com/open-metadata/OpenMetadata/assets/22651118/c559fcd2-9403-4bd4-bc02-d797e9bc88af)

**Describe the solution you'd like**
Today OMD relies on the nodes segment of Tableau metadata to create the data model.

```
  embeddedDatasourcesConnection(first: {first}, offset: {offset} ) {{
    nodes {{
      id
      name
      fields {{
        id
        name
        upstreamColumns{{
          id
          name
          remoteType
        }}
```

But perhaps a better way would be to create the data model based on the root data model, since these share the same ID across the models
![image](https://github.com/open-metadata/OpenMetadata/assets/22651118/04ac6fca-e0f5-4904-a830-d60f296de5d5)

**Comment by chillerno1** [comment]
@harshach I wanted to chime in on this conversation. At my organisation, we're ingesting a large Tableau instance and we've also noticed this behavior where there are multiple _versions_ of the same datasource. What we found, is that a workbook (dashboard) can have it's own embedded datasource, which is a workbook unique version of an upstream datasource that it connects to (usually one that exists on Tableau server). The reason for this, seems to be, that a workbook can connect to a datasource, then change field names, add calculated fields and do various other things to have it's own version of the connected datasource. 

Ideally what we would like (and @jsampaiog please jump in if you disagree); is for published datasources to be ingested into OpenMetadata as well as the embedded datasources (would be nice to have a different icon to differentiated the different datamodels). 

I think it's important to keep both the published and embedded datasources, because that way we can see what transformations have occurred at the workbook level and compare it to the published server model.

Here's a screenshot of what it might look like:

![image](https://github.com/open-metadata/OpenMetadata/assets/17304934/ea453f2b-aa8c-4310-88c4-684f08a1b374)

**Comment by jsampaiog** [comment]
Hi @chillerno1, thanks for chiming in. Indeed your depicted behavior would be the best target scenario!

**Comment by chillerno1** [comment]
Thanks @jsampaiog, I agree with that!

**Comment by nicor88** [comment]
@pmbrull are you still planning to include this in 1.4.0 release? I see that was removed :(

**Comment by pmbrull** [comment]
hi we had to reprioritize certain topics and ran out of time to handle this, so 1.4.1 - 1.5 would be the new ETA.

My 2 cents on the conversation above is to keep things simple. Aiming to keep the `Published DataModel` IMO would be the way to go to reduce complexity

**Comment by nicor88** [comment]
@pmbrull thanks for the context on the timelines.

I believe that "Published DataModel" should do the job even in case of "Dashboard" with embedded data models. We just need to be sure that we don't introduce a regression, where data models are totally missing.

**Comment by 17rahulsharma** [comment]
Thanks @OnkarVO7 , for the this thread.


We also have similar problem of having duplicate Models rather a combined model for all workbooks down the stream.

Since currently OMD use this query

query  {
  embeddedDatasourcesConnection(filter: {name: "Tech Data Model"}) {
    nodes{
      id
    	name
      workbook {
        id
        name
      }
    }
    totalCount
  }
}

We checked with Tableau team (spent a lot of time with Tableau support team to get information in right way) and they proposed to use below query 

query  {
  publishedDatasources(filter: {name: "Tech Data Model"}) {
    id
    name
    hasExtracts
    downstreamWorkbooks{
      id
      luid
      name
    }
  }
}

**Comment by triquinielas** [comment]
Hi, 

Sorry for commenting in this thread, we are facing the same situation: the sources are duplicated for each workbook (dashboard in OM) that we ingested.

![image](https://github.com/open-metadata/OpenMetadata/assets/139678112/150bec44-26f8-4cc6-b04c-7d3d4be270f6)

The dashboard datamodel exists only once on Tableau:
![image](https://github.com/open-metadata/OpenMetadata/assets/139678112/aa56068b-d13b-48b0-9373-752f37bec981)

![image](https://github.com/open-metadata/OpenMetadata/assets/139678112/aac00600-8989-436f-b2da-2535872b792e)

In addition, we have it separated by different services, each service is a tableau folder, since this allows us to assign owner by folder, perhaps, if in the ingestion the folder (tableau) is ingested as the database service would allow us to maintain that hierarchy that also allows us to filter by folder:

DB Ingestion->Schema->table
Ingestion Tableau->**_Folder_**->Workbook & datamodels

thanks,
Carlos

**Comment by nicor88** [comment]
Do I get this right, that using the query provided in here https://github.com/open-metadata/OpenMetadata/issues/15218#issuecomment-2084583097 might be enough to cover this issue?

**Comment by luaguirre1983** [comment]
Hi! A quick question. Was it finally fix this issue in version 1.5? I'd like to understand if yes, so start adding the new version
